tests/Agent/IntegrationTests/IntegrationTests/AgentFeatures/VulnerabilityManagementTests.cs

```diff
@@ -0,0 +1,117 @@
+    {
+        private const string Namespace = "namespace";
+        private const string PublicKeyToken = "publicKeyToken";
+        private const string Sha1Checksum = "sha1Checksum";
+        private const string Sha512Checksum = "sha512Checksum";
+        private const string AssemblyHashCode = "assemblyHashCode";
+        private const string ImplementationVendor = "Implementation-Vendor";
+        private const string Copyright = "copyright";
+        private const string MicrosoftName = "Microsoft";
+
+        protected readonly TFixture _fixture;
+
+        public VulnerabilityManagementTestsBase(TFixture fixture, ITestOutputHelper output) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+            _fixture.SetTimeout(TimeSpan.FromMinutes(3));
+
+            // the commands doesn't really matter as long as the agent starts and runs correctly.
+            _fixture.AddCommand($"ApiCalls TestGetLinkingMetadata");
+
+            _fixture.AddActions
+            (
+                setupConfiguration: () =>
+                {
+                    var configModifier = new NewRelicConfigModifier(fixture.DestinationNewRelicConfigFilePath);
+                    configModifier.SetLogLevel("debug");
+                    configModifier.ConfigureFasterUpdateLoadedModulesCycle(15);
+                },
+                exerciseApplication: () =>
+                {
+                    _fixture.AgentLog.WaitForLogLine(AgentLogBase.AgentConnectedLogLineRegex, TimeSpan.FromMinutes(2));
+                    _fixture.AgentLog.WaitForLogLine(AgentLogBase.UpdateLoadedModulesLogLineRegex, TimeSpan.FromMinutes(2));
+                }
+            );
+
+            _fixture.Initialize();
+        }
+
+        [Fact]
+        public void Test()
+        {
+            var assemblies = _fixture.AgentLog.GetUpdateLoadedModulesAeemblies();
+
+            // This assembly is always present and has all the items we send so we can verify that we capture everything.
+            var mscorlib = assemblies.First(a => a.AssemblyName == "mscorlib.dll");
+
+            // We can be only certain of a few bits of information, like the AssemblyName and Namespace.
+            // We can check the contents of those items, but for the others we just want to make sure there is data.
+            NrAssert.Multiple
+            (
+                () => Assert.NotEmpty(assemblies),
+                () => Assert.Equal("mscorlib.dll", mscorlib.AssemblyName),
+                () => Assert.False(string.IsNullOrWhiteSpace(mscorlib.AssemblyVersion)),
+                () => Assert.Equal(7, mscorlib.Data.Count),
+
+                () => Assert.True(mscorlib.Data.ContainsKey(Namespace)),
+                () => Assert.Equal("mscorlib", mscorlib.Data[Namespace]),
+
+                () => Assert.True(mscorlib.Data.ContainsKey(PublicKeyToken)),
+                () => Assert.False(string.IsNullOrWhiteSpace(mscorlib.Data[PublicKeyToken])),
```

Comment:
You might be able to simplify this by using one of the methods available on StringAssert.

Reply:
StringAssert is not used in the integration tests and the package, MsTest.TestFramework,  is not installed.

Reply:
Thanks for checking. I couldn't remember which library those helper assertions belonged to.